app/questionnaire/questionnaire_schema.py

```diff
@@ -714,7 +723,7 @@ def get_unrelated_block_no_answer_values(
             ]
 
     @staticmethod
-    def get_single_string_value(schema_object: Union[Mapping, str]) -> str:
+    def get_single_string_value(schema_object: Mapping | str) -> str:
```

Comment:
And these mappings here and line 858 need top level arguments, same with any sequences.